### PR TITLE
Transfer: add support for outputs import / export

### LINKED ISSUE #2816: Support for importing and exporting outputs
In a setup with an offline private-key wallet and an online view-only wallet, one needs to import key images from the offline wallet to the view-only wallet in order to see the true balance that takes into account spending. But the offline wallet needs to know the outputs for it to be able to create the key images. So, one needs to be able to export the outputs from the view-only wallet and import them to the offline wallet. However, GUI doesn't support these two operations. CLI supports `export_outputs` and `import_outputs`. It would be nice to have these features available in the GUI too.

The source of my information was this dicussion:

https://www.reddit.com/r/Monero/comments/fra1k5/questions_on_key_images/flusnou/

Which gave the following instructions:

- Import address, private view key into watch only wallet. Scan for incoming transactions.

- Export the list of outputs from the watching wallet to the offline wallet.

- Import the outputs into the offline wallet, generate key images, export key images to watching wallet.

- Watching wallet is now ready to create unsigned transactions for the offline wallet to sign.

### PR DESCRIPTION
**Pull request by selsta**
Requires https://github.com/monero-project/monero/pull/7678

Resolves #2816

### CODE REVIEW

File: pages/Transfer.qml [hunk 1]
```diff
@@ -843,6 +843,32 @@ Rectangle {
...
+                exportOutputsDialog.open();
+            }
+            button2.text: qsTr("Import") + translationManager.emptyString
+            button2.enabled: !appWindow.viewOnly
+            button2.onClicked: {
+                console.log("Transfer: import outputs clicked")
+                importOutputsDialog.open();
+            }
+            tooltip: {
+                var header = qsTr("Required for cold wallets to to sign their corresponding key images") + translationManager.emptyString;
```
Comment (rating89us) [inline pages/Transfer.qml:862]:
```suggestion
                var header = qsTr("Required for cold wallets to sign their corresponding key images") + translationManager.emptyString;
```

Comment (selsta) [reply pages/Transfer.qml]: oops, thanks :)

**Comment by selsta** [comment]
@rating89us please review / approve

**Comment by rating89us** [comment]
I'm getting this error:
`W qrc:/pages/Transfer.qml:1107: TypeError: Type error`

**Comment by selsta** [comment]
@rating89us how did you get this message?

**Comment by rating89us** [comment]
On view-only wallet, I clicked the export button. A file browser window opened and I entered a filename, when I click on the save button, no file was created and this error message appears on console.
I'm using Ubuntu.

**Comment by rating89us** [comment]
I didn't import https://github.com/monero-project/monero/pull/7678 before testing it. Is it included in master?

**Comment by selsta** [comment]
> I didn't import monero-project/monero#7678 before testing this PR. Is it included in master?

It's included in v0.17.2.3 and master. Will test in view only wallet.

**Comment by selsta** [comment]
@rating89us Thanks, found an issue. Was able to export fine now.

**Comment by rating89us** [comment]
1) Output exporting is working now on view-only wallet. I suggest we add a feedback for the user:
`appWindow.showStatusMessage(qsTr("Outputs successfully exported to file") + translationManager.emptyString, 3);`

2) When importing the outputs file in cold wallet, I get this error:
`W qrc:/pages/Transfer.qml:1121: TypeError: Property 'importOutputsDialog' of object Wallet(0x7f342a9d7b20) is not a function`

**Comment by selsta** [comment]
> Output exporting is working now on view-only wallet. I suggest we add a feedback for the user:

Fixed

> When importing the outputs file in cold wallet, I get this error:

Fixed.. had copy paste error

**Comment by selsta** [comment]
Updated on more time to print success / error message.

**Comment by rating89us** [comment]
Now it's working for me.

I think we should also print success / error message for importing/exporting key images.

**Review by rating89us — APPROVED** [review]

**Comment by selsta** [comment]
Also updated error string here.

**Review by rating89us — APPROVED** [review]

**Pull request merged**